packages/gallery/src/components/item/itemHover.js

```diff
@@ -54,17 +54,107 @@ export default class ItemHover extends GalleryComponent {
     return true;
   }
 
+  getOverlayStyle() {
+    const { styleParams } = this.props;
+    const { overlayPosition } = styleParams;
+    const isHorizontal =
+      overlayPosition === GALLERY_CONSTS.overlayPositions.LEFT ||
+      overlayPosition === GALLERY_CONSTS.overlayPositions.RIGHT ||
+      overlayPosition === GALLERY_CONSTS.overlayPositions.CENTERED_HORIZONTALLY;
+    if (isHorizontal) {
+      return this.calculateHorizontalOverlayStyles();
+    }
+    return this.calculateVerticalOverlayStyles();
+  }
+
+  calculateVerticalOverlayStyles() {
+    const { styleParams, imageDimensions } = this.props;
```

Comment:
Remove calculations outside of the component. All these calculations are dependant on a few props (from the `styleParams`) therefor should not be performed inside a component that renders